### PR TITLE
overlord: fix broken tests

### PR DESCRIPTION
**Pull request by zyga**
This branch exposes ``SnapState``, ``GetSnapState`` and ``SetSnapState`` and uses them in the interfaces test so that everything works correctly.

### CODE REVIEW

File: overlord/snapstate/export_test.go [hunk 1]
```diff
@@ -45,6 +43,6 @@ func (m *SnapManager) AddForeignTaskHandlers() {
 	m.runner.AddHandler("remove-snap-security", fakeHandler, fakeHandler)
 }
 
-func SetSnapState(st *state.State, name string, ss *SnapStateForTests) {
-	setSnapState(st, name, (*snapState)(ss))
+func SetSnapState(st *state.State, name string, ss *SnapState) {
+	setSnapState(st, name, (*SnapState)(ss))
```
Comment (pedronis) [inline overlord/snapstate/export_test.go]: we don't need the cast anymore here

Comment (zyga) [inline overlord/snapstate/export_test.go]: We don't need the whole function

**Comment by mvo5** [comment]
Looks good

**Comment by pedronis** [comment]
+1

**Pull request merged**